lib/helpers.js

```diff
@@ -1,19 +1,40 @@
+const memoize = require('memoizerific');
+
 const assert = (condition, msg = 'assertion failed') => {
   if (!condition) {
     throw new Error(msg);
   }
 };
 
 // FIXME: use in `match`.
-function replaceAll(str, char, replacement = '') {
+/**
+ * Replace all occurences of replacee in str.
+ * @param {string} str
+ * @param {string} replacee
+ * @param {string} replacement
+ * @returns {string}
+ */
+function replaceAll(str, replacee, replacement = '') {
   let _str;
   do {
     _str = str;
-    str = str.replace(char, replacement);
-  } while (str != _str);
+    str = str.replace(replacee, replacement);
+  } while (str !== _str);
   return str;
 }
 
+/**
+ * Replace all occurences of codepoints in str.
+ * @param {string} str
+ * @param {string[]} codepoints
+ * @param {string} replacement
+ * @returns {string}
+ */
+function replaceAllCodepoints(str, codepoints, replacement = '') {
```

Comment:
When the thing to be replaced is an array, it's faster to make into a global regex than to loop until the target string isn't changing anymore.